### PR TITLE
support Terapines zcc toolchain

### PR DESCRIPTION
**Pull request by kiorito**

### CODE REVIEW

**Comment by jhqian** [comment]
Thanks for this pull request, as we're not taking any PR right now. But we'll integrate it into sdk 1.6.0 which might be modified a little bit. Could you please provide your mail? we'll keep your info in corresponding commit message.

**Comment by kiorito** [comment]
> Thanks for this pull request, as we're not taking any PR right now. But we'll integrate it into sdk 1.6.0 which might be modified a little bit. Could you please provide your mail? we'll keep your info in corresponding commit message.

Sure, the mail is jingjing.huang@terapines.com, thanks.

File: cmake/ide/segger.cmake [hunk 1]
```diff
@@ -196,6 +196,12 @@ function (generate_ses_project)
         get_property(target_nds_cflags TARGET ${HPM_SDK_NDSGCC_LIB_ITF} PROPERTY INTERFACE_COMPILE_OPTIONS)
     endif()
 
+    string(FIND ${SES_TOOLCHAIN_VARIANT} "Terapines" exist)
```
Comment (jhqian) [inline cmake/ide/segger.cmake:199]: Is SES going to support ZCC officially? What's the timeline?

File: soc/HPM6280/CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +32,22 @@ endif()
 
 sdk_nds_compile_options(-mcpu=d45)
 
+if("${TOOLCHAIN_VARIANT}" STREQUAL "zcc")
+    sdk_compile_options(--target=riscv32-unknown-elf)
+    sdk_compile_options(-march=rv32imafdcxandes)
+    sdk_compile_options(-mabi=ilp32d)
+    sdk_compile_options(-mtune=andes-kavalan)
```
Comment (jhqian) [inline soc/HPM6280/CMakeLists.txt:39]: What's this option (andes-kavalan) for? Is it only for D45?

File: soc/HPM6280/CMakeLists.txt [hunk 3]
```diff
@@ -32,6 +32,22 @@ endif()
 
 sdk_nds_compile_options(-mcpu=d45)
 
+if("${TOOLCHAIN_VARIANT}" STREQUAL "zcc")
+    sdk_compile_options(--target=riscv32-unknown-elf)
+    sdk_compile_options(-march=rv32imafdcxandes)
+    sdk_compile_options(-mabi=ilp32d)
```
Comment (jhqian) [inline soc/HPM6280/CMakeLists.txt:38]: FPU will cause significant isr latency due to context saving, is it any specific reason to enable FPU by default?

**Comment by kiorito** [comment]
Sorry, I closed this #31 by mistake, so I pull a new request #34.

**Pull request closed without merging**